Base/QTApp/Resources/UI/qSlicerErrorReportDialog.ui

```diff
@@ -114,7 +114,7 @@
     </layout>
    </item>
    <item row="2" column="0" colspan="2">
-    <widget class="ctkPathListWidget" name="RecentLogFilesComboBox"/>
+    <widget class="QListWidget" name="RecentLogFilesComboBox"/>
```

Comment:
> Please also add the application name (or whatever the unprocessed string is that precedes the parsed revision, date, and time). 


It appears the log files that are available to choose from are only those from "%TEMP%\ApplicationName". Therefore won't every log in the table be from the same application name and therefore not adding value to the UI here?

Reply:
Most of the time the prefix will be the application name, but indeed any module can put additional log files in that folder. These additional filenames are not even guaranteed to follow the filename pattern (appname_rev_date_time_pid.log), so the table content should be reasonable in this case, too (would be nice to do a basic check, on the filename components).

Reply:
I see now that the list is populated from recent application log files, so indeed only the application log files are included in the list.

Somehow I still miss seeing the application name there, as it feels odd to see a revision number without context (without seeing what it is the revision of).

Reply:
I could definitely see the case of including application name of Slicer custom apps shared the same log directory but that doesn’t seem to be the case.

Even revision isn’t quite “human readable” in the sense that non-developers do not understand what this means. More readable would be the Application Version (with date for Preview builds). Then users could easily know if the log entry is from say Slicer 5.4.0 vs 5.2.2.

Reply:
Adding the application version as well would be nice, @nhjohnston please add this to `SlicerApplication::setupFileLogging()`. You need to make the parsing compatible with both the old filename format (no application version) and new format (including application version).

It would be nice if you could experiment with adding a column for the application name and the full path of the filename (it could be the last column, which takes up the remaining space in the table) and take a few screenshots.

Reply:
Here is a screenshot using `QTableWidget` and added the application version to the log file name. I am working on parsing the dates/times to be more human-readable, using QString rather than STL, but wanted to post the screenshot for initial feedback. Currently, I removed the file icon because I was not sure which cell to put it in - is there a preference?
![image](https://github.com/Slicer/Slicer/assets/88200986/1b36ce18-f02f-4560-8692-ab650a7f3d1b)

Reply:
I think this table view will look nice. Configure full-row selection in the table (so that the full row is selected and not just a cell) and tune the column sizing beheavior (all can be resize to content, maybe except the last file path column).

The long file path in the last column may make the table look more messy by making rows taller and/or adding a horizontal scrollbar. If we find that we cannot tune the table options so that the table looks nice and clean then maybe we can remove the file path column and make double-click action copy the path to the clipboard (the main disadvantage that it would be hard to discover this, but maybe it would help showing a "File path copied to the clipboard." popup for 1-2 seconds and/or adding a note about double-click to the tooltip) or we coud display the log file path next to the "Log file content:" label in a textbox (from where user could copy to the clipboard).

Reply:
Here are variations on the column sizing - with date/time parsing as recommended by @pieper .

| Path Column not Sized to Contents | All Columns Sized to Contents |
|------------|----------|
|![image](https://github.com/Slicer/Slicer/assets/88200986/115df86a-cefc-46af-b066-597440afd31d)|![image](https://github.com/Slicer/Slicer/assets/88200986/eed1f99d-c95f-47bc-8878-72ac9c353776)|